### ISSUE TITLE
Some dependencies are not resolved when declaring repositories in settings

### ISSUE THREAD

**Issue by Corje** [Kotlin/kotlinx-kover#9]
Create a project with root and child module (for example simple android application).
Declare repositories in settings file and set option to prefer settings repositories over subprojects (see [gradle doc](https://docs.gradle.org/current/userguide/declaring_repositories.html#sub:centralized-repository-declaration)):
```groove
dependencyResolutionManagement {
    repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)
    repositories {
        google()
        mavenCentral()
        maven { url 'https://maven.pkg.jetbrains.space/public/p/jb-coverage/maven' }
    }
}
```
In child module apply kover plugin. Run test task.
There is a warning:
```
> Configure project :app
Build was configured to prefer settings repositories over project repositories but repository 'maven' was added by plugin 'kotlinx-kover'
Build was configured to prefer settings repositories over project repositories but repository 'maven2' was added by plugin 'kotlinx-kover'
```
And then error:
```
* What went wrong:
Execution failed for task ':app:testDebugUnitTest'.
> Could not resolve all files for configuration ':app:IntellijKoverConfig'.
   > Could not find org.jetbrains.intellij.deps:intellij-coverage-agent:1.0.608.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/intellij/deps/intellij-coverage-agent/1.0.608/intellij-coverage-agent-1.0.608.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/intellij/deps/intellij-coverage-agent/1.0.608/intellij-coverage-agent-1.0.608.pom
       - https://maven.pkg.jetbrains.space/public/p/jb-coverage/maven/org/jetbrains/intellij/deps/intellij-coverage-agent/1.0.608/intellij-coverage-agent-1.0.608.pom
       - https://jcenter.bintray.com/org/jetbrains/intellij/deps/intellij-coverage-agent/1.0.608/intellij-coverage-agent-1.0.608.pom
     Required by:
         project :app
   > Could not find org.jetbrains.intellij.deps:coverage-report:1.0.7.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/intellij/deps/coverage-report/1.0.7/coverage-report-1.0.7.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/intellij/deps/coverage-report/1.0.7/coverage-report-1.0.7.pom
       - https://maven.pkg.jetbrains.space/public/p/jb-coverage/maven/org/jetbrains/intellij/deps/coverage-report/1.0.7/coverage-report-1.0.7.pom
       - https://jcenter.bintray.com/org/jetbrains/intellij/deps/coverage-report/1.0.7/coverage-report-1.0.7.pom
     Required by:
         project :app > org.jetbrains.intellij.deps:intellij-coverage-reporter:0.255
```
In case `RepositoriesMode` in `settings.gradle` is `FAIL_ON_PROJECT_REPOS` build just fails with:
```
An exception occurred applying plugin request [id: 'kotlinx-kover', version: '0.2.2']
> Failed to apply plugin 'kotlinx-kover'.
   > Build was configured to prefer settings repositories over project repositories but repository 'maven' was added by plugin 'kotlinx-kover'
```

Looks like implicit repositories declared by kover plugin break the build.

The issue is solved when you just declare repositories in submodule but not in settings.

Sample project is attached
[AndroidCoverage.zip](https://github.com/Kotlin/kotlinx-kover/files/7259233/AndroidCoverage.zip)

**Comment by shanshin** [comment]
Duplicate #34

**Comment by shanshin** [comment]
Fixed in `0.4.3`
